src/node_dotenv.cc

```diff
@@ -59,50 +64,42 @@ std::vector<Dotenv::env_file_data> Dotenv::GetDataFromArgs(
   return env_files;
 }
 
-void Dotenv::SetEnvironment(node::Environment* env) {
-  auto isolate = env->isolate();
+Maybe<void> Dotenv::SetEnvironment(node::Environment* env) {
+  Local<Value> name;
+  Local<Value> val;
+  auto context = env->context();
 
   for (const auto& entry : store_) {
-    auto key = entry.first;
-    auto value = entry.second;
-
-    auto existing = env->env_vars()->Get(key.data());
-
+    auto existing = env->env_vars()->Get(entry.first.data());
     if (!existing.has_value()) {
-      env->env_vars()->Set(
-          isolate,
-          v8::String::NewFromUtf8(
-              isolate, key.data(), NewStringType::kNormal, key.size())
-              .ToLocalChecked(),
-          v8::String::NewFromUtf8(
-              isolate, value.data(), NewStringType::kNormal, value.size())
-              .ToLocalChecked());
+      if (!ToV8Value(context, entry.first).ToLocal(&name) ||
+          !ToV8Value(context, entry.second).ToLocal(&val)) {
+        return Nothing<void>();
+      }
+      env->env_vars()->Set(env->isolate(), name.As<String>(), val.As<String>());
     }
   }
+
+  return JustVoid();
 }
 
-Local<Object> Dotenv::ToObject(Environment* env) const {
+MaybeLocal<Object> Dotenv::ToObject(Environment* env) const {
+  EscapableHandleScope scope(env->isolate());
```

Comment:
For educational purposes: why is this needed?

Reply:
Any JS handles that are created while this is running will be created within whatever handlescope is on the stack. If this method errors, those handles will end up being kept around until that outer scope exits. This ensures that in the error case, any handles created during this function are cleaned up immediately.

Reply:
Why do we not have this everywhere?

Reply:
It's really not needed everywhere. Whether it is needed here or not is debatable.